client/src/components/game/GamePage.jsx

```diff
@@ -29,6 +29,11 @@ function GamePage() {
 
         gameInstance.scene.start("Boot");
 
+        window.onbeforeunload = () => {
+            PubSub.publish(BEFORE_PAGE_UNLOAD);
+            return ("");
```

Comment:
I don't see any implementation of detecting environment now, but this one should be disabled during development.

something like this:
```js
PubSub.publish(BEFORE_PAGE_UNLOAD);
if (ENVIRONMENT !== "development") {
  return "";
}
```

*I have a feeling that this will be annoying when developing new feature as you will be bombarded with alerts every time you made a change in the code.*

Reply:
Yeah. This code sets the `dungeonz.devMode` flag that can be used here. (Though it probably would now be better somewhere like the Application state)

```js
if(!dungeonz.devMode){
    window.onbeforeunload = () => {
        PubSub.publish(BEFORE_PAGE_UNLOAD);
        return ("");
    }
}
```